core/src/main/java/org/infinispan/CacheStreamable.java

```diff
@@ -0,0 +1,120 @@
+package org.infinispan;
+
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.Set;
+import java.util.Spliterator;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+import java.util.stream.BaseStream;
+
+/**
+ * Interface that signifies which streams are able to be used with a {@link Cache}.  This method
+ * is useful to hold a reference to any of the types while still being able to invoke some methods.
+ */
+public interface CacheStreamable<T, S extends BaseStream<T, S>> extends BaseStream<T, S> {
```

Comment:
I'm not sure about the name, to me it suggests something that has a `stream()` method. Couldn't we follow the JDK's lead and name it `BaseCacheStream`?